### ISSUE TITLE
Feature request: implement sigsuspend

### ISSUE THREAD

**Issue by germandiagogomez** [emscripten-core/emscripten#19880]
Hello emscripten developers,

I am on my way to port some code to the web for a client-side application. My application is using Cap'n Proto serialization from C++: https://capnproto.org/

My code is almost fully compiling under emscripten except for the lack of `sigsuspend` implementation.

Any chances it will be implemented?

Alternatively, any hints on how to implement it? I am not very knowledgeable of the inner workings of the web platform itself.

**Comment by germandiagogomez** [comment]
Is there any possibility that sigsuspend will be supported or a way for me to implement it? Any would be welcome. 

Thanks.

**Comment by vadimkantorov** [comment]
@sbc100 would it be a good approach to introduce a separate stub library which would include stub impls for currently-not-implemented functions? (so that compilation can procede, especially if we know that that code path will not be taken ever)

**Comment by sbc100** [comment]
Yes, we do have a place for such stubs already: https://github.com/emscripten-core/emscripten/blob/main/system/lib/libc/emscripten_libc_stubs.c

**Comment by sbc100** [comment]
We have a setting (enabled by default) called `ALLOW_UNIMPLEMENTED_SYSCALLS`:  https://emscripten.org/docs/tools_reference/settings_reference.html#allow-unimplemented-syscalls.

You can disabled this yourself, and it is also off by default in `-sSTRICT` mode.

In this case sigsuspend is already in the musl headers, and it just needs to be added to emscripten_libc_stubs.c.  I guess that only sensible default is to abort.. or maybe return -1 with `ENOSYS` errno set?

**Comment by vadimkantorov** [comment]
Yeah, maybe `return -1 with ENOSYS errno set`. But in practice, it also would depend on how autoconf checks do detection typically. If they check actual workability, then they would rightfully fail the check, but at least it's not the nasty linking warning.
